microsoftgraph/docs-conceptual/tutorial-grant-app-only-api-permissions.md

```diff
@@ -0,0 +1,145 @@
+
+## Step 1: Get the app roles of the resource service principal
+
+Before you can grant app roles, you must first identify the app roles to grant and the resource service principal that exposes the app roles. App roles are defined in the `appRoles` object of a service principal. 
+
+In this article, you'll use the `Microsoft Graph` service principal in the tenant as your resource service principal.
+
+```powershell
+Get-MgServicePrincipal -Filter "displayName eq 'Microsoft Graph'" -Property AppRoles | Select -ExpandProperty appRoles |fl
+```
+
+```Output
+AllowedMemberTypes   : {Application}
+Description          : Allows the app to read and update user profiles without a signed in user.
+DisplayName          : Read and write all users' full profiles
+Id                   : 741f803b-c850-494e-b5df-cde7c675a1ca
+IsEnabled            : True
+Origin               : Application
+Value                : User.ReadWrite.All
+AdditionalProperties : {}
+
+AllowedMemberTypes   : {Application}
+Description          : Allows the app to read user profiles without a signed in user.
+DisplayName          : Read all users' full profiles
+Id                   : df021288-bdef-4463-88db-98f22de89214
+IsEnabled            : True
+Origin               : Application
+Value                : User.Read.All
+AdditionalProperties : {}
+```
+
+## Step 2: Create a client service principal
+
+The first step in granting consent is to [create the service principal](/powershell/module/microsoft.graph.applications/new-mgserviceprincipal?view=graph-powershell-1.0&preserve-view=true). To do so, you'll need the `App Id` of your application.
+
+### Register an application with Azure AD
+
+If the application is not available, register an application with Azure AD.
+
+```powershell
+New-MgApplication -DisplayName 'My application' | 
+  Format-List Id, DisplayName, AppId, SignInAudience, PublisherDomain
+```
+
+```Output
+Id              : 40cbfad6-f138-4fb4-9e7f-5a44044efbd6
+DisplayName     : My application
+AppId           : 05210c44-437f-4a40-bd38-b5b4eaf251ef
+SignInAudience  : AzureADandPersonalMicrosoftAccount
+PublisherDomain : Contoso.com
+```
+
+### Create a service principal for the application
+
+```powershell
+New-MgServicePrincipal -AppId '05210c44-437f-4a40-bd38-b5b4eaf251ef' | 
+  Format-List Id, DisplayName, AppId, SignInAudience
+```
+
+```Output
+Id             : 22c1770d-30df-49e7-a763-f39d2ef9b369
+DisplayName    : My application
+AppId          : 05210c44-437f-4a40-bd38-b5b4eaf251ef
+SignInAudience : AzureADandPersonalMicrosoftAccount
+```
+
+## Step 3: Assign an app role to the client service principal
+
+In this step, you'll assign an app role exposed by your resource app to the service principal we created in step 2. To create an app role assignment, you'll need the following information:
+
+1. **PrincipalId** - object Id of the service principal authorized for direct access.
```

Comment:
**PrincipalId** - object Id of the service principal *to be* authorized for direct access.? Just a thought.